src/components/network/ImportNetwork.tsx

```diff
@@ -0,0 +1,137 @@
+import styled from '@emotion/styled';
+import { Button, Upload, Card, PageHeader, Spin, Modal } from 'antd';
+import { UploadOutlined } from '@ant-design/icons';
+import log from 'electron-log';
+import { ThemeColors } from 'theme/colors';
+import { useTheme } from 'hooks/useTheme';
+import { HOME } from 'components/routing';
+import { useStoreActions, useStoreState } from 'store';
+import { usePrefixedTranslation } from 'hooks';
+import { RcFile } from 'antd/lib/upload';
+import { getNetworkFromZip } from 'utils/network';
+import fsExtra from 'fs-extra';
+import { promises as fs } from 'fs';
+import { join } from 'path';
+import { dataPath } from 'utils/config';
+
+const Styled = {
+  PageHeader: styled(PageHeader)<{ colors: ThemeColors['pageHeader'] }>`
+    border: 1px solid ${props => props.colors.border};
+    border-radius: 2px;
+    background-color: ${props => props.colors.background};
+    margin-bottom: 10px;
+    flex: 0;
+  `,
+  ButtonContainer: styled.div`
+    margin-top: 20px;
+    display: flex;
+    justify-content: space-evenly;
+
+    button {
+      width: 200px;
+    }
+  `,
+  Spin: styled(Spin)<{ visible: boolean }>`
+    position: absolute;
+    display: ${visible => (visible ? 'inherit' : 'none')};
+  `,
+};
+
+const ImportNetwork: React.SFC = () => {
+  const [file, setFile] = useState<RcFile | undefined>();
+  const { navigateTo } = useStoreActions(s => s.app);
+  const { l } = usePrefixedTranslation('cmps.network.ImportNetwork');
+  const networkActions = useStoreActions(s => s.network);
+  const designerActions = useStoreActions(s => s.designer);
+
+  const { networks } = useStoreState(s => s.network);
+
+  const theme = useTheme();
+  return (
+    <>
+      <Styled.PageHeader
+        title={l('title')}
+        colors={theme.pageHeader}
+        onBack={() => navigateTo(HOME)}
+      />
+      <Card>
+        <Upload.Dragger
+          fileList={file ? [file] : []}
+          accept=".zip"
+          onRemove={async () => {
+            setFile(undefined);
+
+            // return false makes the operation stop. we don't want to actually
+            // interact with a server, just operate in memory
+            return false;
+          }}
+          beforeUpload={file => {
+            setFile(file);
+
+            // return false makes the upload stop. we don't want to actually
+            // upload this file anywhere, just store it in memory
+            return false;
+          }}
+        >
+          <p className="ant-upload-drag-icon">
+            <UploadOutlined />
+          </p>
+          <p className="ant-upload-text">{l('fileDraggerArea')}</p>
+        </Upload.Dragger>
+        <Styled.ButtonContainer>
+          <Button onClick={() => setFile(undefined)} disabled={file === undefined}>
+            {l('removeButton')}
+          </Button>
+          <Button
+            onClick={async () => {
+              if (!file) {
+                throw Error('File was undefined in import submit function!');
```

Comment:
If this error is thrown it will crash the app, it would be to display an error notification instead.

Reply:
This is more of an invariant that should always hold, though. If the file is undefined, the button should be disabled. So if this point is ever reached, we have a bug. I think it's good practice to be explicit about errors that should never happen (as opposed to errors that could very well happen, these should be handled gracefully). What's your thoughts on this?

Reply:
I would argue that throwing errors for situations that should never happen is a bit misleading. The fact that you are explicitly checking for this case makes me believe that it is possible when reading the code. Using this conditional also means you should have a unit test to exercise this logic, which would be difficult to do since the button is disabled. I think it's better to just cast the object `file as RcFile` to make Typescript happy. As someone reading the code, the cast tells me that `file` should never be undefined, which is true.